### PR TITLE
illumos-userland 2734 - add libgit2

### PR DESCRIPTION
**Pull request by rtvd**

### CODE REVIEW

File: components/library/libgit2/libgit2.p5m [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+dir  path=usr
```
Comment (pyhalov) [inline components/library/libgit2/libgit2.p5m]: remove dir actions

File: components/library/libgit2/Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+CMAKE_OPTIONS+=			-DCMAKE_C_COMPILER=gcc
+
+build:		$(BUILD_32_and_64)
+
+install:	$(INSTALL_32_and_64)
+
+test:		$(TEST_32_and_64)
+
+include $(WS_MAKE_RULES)/depend.mk
```
Comment (pyhalov) [inline components/library/libgit2/Makefile]: No need to include depend.mk.
Please, add REQUIRED_PACKAGES

**Comment by rtvd** [comment]
More changes as a follow up for the comments above.

File: components/library/libgit2/libgit2.p5m (hunk 1 above)
Comment (rtvd) [inline components/library/libgit2/libgit2.p5m]: Thank you. Removed them, except for those which are specific to the library. Surely directories like /usr/include/git2 should stay, should they not?

Comment (alarcher) [inline components/library/libgit2/libgit2.p5m]: Directory actions are automatically generated if file actions require it i.e. there exists a file delivered in given path.
Directories should only be left in the manifest if no content is provided upon installation e.g. logging directories.

File: components/library/libgit2/libgit2.p5m [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+dir  path=usr/include/git2
```
Comment (xen0l) [inline components/library/libgit2/libgit2.p5m]: Remove dir actions

File: components/library/libgit2/libgit2.p5m [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+file path=usr/include/git2/revert.h
+file path=usr/include/git2/revparse.h
+file path=usr/include/git2/revwalk.h
+file path=usr/include/git2/signature.h
+file path=usr/include/git2/stash.h
+file path=usr/include/git2/status.h
+file path=usr/include/git2/stdint.h
+file path=usr/include/git2/strarray.h
+file path=usr/include/git2/submodule.h
+dir  path=usr/include/git2/sys
```
Comment (xen0l) [inline components/library/libgit2/libgit2.p5m]: Remove dir actions

File: components/library/libgit2/libgit2.p5m [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+file path=usr/include/git2/trace.h
+file path=usr/include/git2/transaction.h
+file path=usr/include/git2/transport.h
+file path=usr/include/git2/tree.h
+file path=usr/include/git2/types.h
+file path=usr/include/git2/version.h
+link path=usr/lib/$(MACH64)/libgit2.so target=libgit2.so.23
+file path=usr/lib/$(MACH64)/libgit2.so.$(COMPONENT_VERSION)
+link path=usr/lib/$(MACH64)/libgit2.so.23 target=libgit2.so.$(COMPONENT_VERSION)
+dir  path=usr/lib/$(MACH64)/pkgconfig
```
Comment (xen0l) [inline components/library/libgit2/libgit2.p5m]: Remove dir actions

File: components/library/libgit2/libgit2.p5m [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+file path=usr/lib/$(MACH64)/pkgconfig/libgit2.pc
+link path=usr/lib/libgit2.so target=libgit2.so.23
+file path=usr/lib/libgit2.so.$(COMPONENT_VERSION)
+link path=usr/lib/libgit2.so.23 target=libgit2.so.$(COMPONENT_VERSION)
+dir  path=usr/lib/pkgconfig
```
Comment (xen0l) [inline components/library/libgit2/libgit2.p5m]: Remove dir actions

File: components/library/libgit2/Makefile [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+COMPONENT_NAME=		libgit2
+COMPONENT_VERSION=	0.23.4
+COMPONENT_FMRI=		library/libgit2
+COMPONENT_CLASSIFICATION=Development/Source Code Management
+COMPONENT_PROJECT_URL=	https://libgit2.github.com/
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_NAME)-$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:c7f5e2d7381dbc4d7e878013d14f9993ae8a41bd23f032718e39ffba57894029
+COMPONENT_ARCHIVE_URL=	https://github.com/$(COMPONENT_NAME)/$(COMPONENT_NAME)/archive/v$(COMPONENT_VERSION).tar.gz
+COMPONENT_BUGDB=	$(COMPONENT_FMRI)
```
Comment (xen0l) [inline components/library/libgit2/Makefile]: Remove this please

File: components/library/libgit2/Makefile [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+COMPONENT_LICENSE=	GPLv2 with Linking Exception
+COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
+
```
Comment (xen0l) [inline components/library/libgit2/Makefile]: Remove empty line

**Comment by xen0l** [comment]
http-parser also has test Makefile target, so you should add it to component makefile as well.

**Comment by rtvd** [comment]
Is it good now?

**Comment by pyhalov** [comment]
You haven't fixed all issues mentioned by @xen0l

**Comment by rtvd** [comment]
Indeed. Sorry.

**Comment by pyhalov** [comment]
Seems fine, merge commits

**Comment by rtvd** [comment]
Updated REQUIRED_PACKAGES and squashed the commits.

**Comment by rtvd** [comment]
Is it good for the merge?

**Pull request merged**